### PR TITLE
Adding MapCarbonMessage and SerializableCarbonMessage to Carbon messaging

### PR DESCRIPTION
**Pull request by amjadhifthikar**

### CODE REVIEW

File: components/src/main/java/org/wso2/carbon/messaging/MapCarbonMessage.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+     *
+     * @param mapKeyName Key that need to be added to map.
+     * @param mapValue   Value of the key that need to be added to map.
+     */
+    public void setValue(String mapKeyName, String mapValue) {
+        mapData.put(mapKeyName, mapValue);
+    }
+
+    /**
+     * To get the value of a particular key
```
Comment (Kishanthan) [inline components/src/main/java/org/wso2/carbon/messaging/MapCarbonMessage.java]: description should end with a period.

File: components/src/main/java/org/wso2/carbon/messaging/SerializableCarbonMessage.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    private static final long serialVersionUID = 1;
+    private String payload;
+    private String payloadType;
+    private HashMap<String, String> headers;
+
+    public HashMap<String, String> getHeadersMap() {
+        return headers;
+    }
+
+    public void setHeaders(HashMap<String, String> headers) {
```
Comment (Kishanthan) [inline components/src/main/java/org/wso2/carbon/messaging/SerializableCarbonMessage.java]: missing java docs for public methods.

**Review by Kishanthan — CHANGES_REQUESTED** [review]

**Review by Kishanthan — APPROVED** [review]

File: components/src/main/java/org/wso2/carbon/messaging/MapCarbonMessage.java (hunk 1 above)
Comment (amjadhifthikar) [reply components/src/main/java/org/wso2/carbon/messaging/MapCarbonMessage.java]: This is fixed.

File: components/src/main/java/org/wso2/carbon/messaging/SerializableCarbonMessage.java (hunk 2 above)
Comment (amjadhifthikar) [reply components/src/main/java/org/wso2/carbon/messaging/SerializableCarbonMessage.java]: This is fixed.

**Pull request merged**